sawtooth-daml-rpc/src/main/java/com/blockchaintp/sawtooth/daml/rpc/DamlAuthServices.java

```diff
@@ -0,0 +1,166 @@
+ limitations under the License.
+------------------------------------------------------------------------------*/
+package com.blockchaintp.sawtooth.daml.rpc;
+
+import java.math.BigInteger;
+import java.security.AlgorithmParameters;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECGenParameterSpec;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.ECPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.InvalidParameterSpecException;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.digitalasset.daml.lf.data.Ref;
+import com.digitalasset.ledger.api.auth.AuthService;
+import com.digitalasset.ledger.api.auth.AuthServiceJWTPayload;
+import com.digitalasset.ledger.api.auth.Claim;
+import com.digitalasset.ledger.api.auth.ClaimActAsParty$;
+import com.digitalasset.ledger.api.auth.ClaimAdmin$;
+import com.digitalasset.ledger.api.auth.ClaimPublic$;
+import com.digitalasset.ledger.api.auth.Claims;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.spongycastle.util.Arrays;
+
+import akka.protobuf.ByteString;
+import io.grpc.Metadata;
+import scala.collection.immutable.List;
+import scala.collection.immutable.List$;
+import scala.collection.mutable.ListBuffer;
+
+/**
+ * Responsible for decoding JWTToken sent from GRPC
+ *
+ */
+public class DamlAuthServices implements AuthService {
+
+  private final Algorithm ecdsa512Algorithm;
+  private ECPublicKey publicKey;
+
+  public DamlAuthServices(String pubKeyInHex) {
+    byte[] pubKey = ByteString.copyFromUtf8(pubKeyInHex).toByteArray();
+    int midPoint = pubKey.length / 2;
+    byte[] pubKeyX = Arrays.copyOfRange(pubKey, 0, midPoint);
+    byte[] pubKeyY = Arrays.copyOfRange(pubKey, midPoint, pubKey.length);
+    Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+    ECPoint publicPoint = new ECPoint(new BigInteger(1, pubKeyX), new BigInteger(1, pubKeyY));
+    try {
+      AlgorithmParameters parameters = AlgorithmParameters.getInstance("EC");
+      parameters.init(new ECGenParameterSpec("secp256k1"));
+
+      ECParameterSpec params = parameters.getParameterSpec(ECParameterSpec.class);
+      ECPublicKeySpec pubKeySpec = new ECPublicKeySpec(publicPoint, params);
+
+      KeyFactory kf = KeyFactory.getInstance("EC");
+      this.publicKey = (ECPublicKey) kf.generatePublic(pubKeySpec);
+
+      this.ecdsa512Algorithm = Algorithm.ECDSA512(this.publicKey, null);
```

Comment:
It would be worth-while to push this algorithm upstream as an alternative JwtVerifier
https://github.com/digital-asset/daml/blob/878429e3bf07b09e727224d5dc423444d071a95b/ledger-service/jwt/src/main/scala/com/digitalasset/jwt/JwtVerifier.scala